src/common/Widgets/VpnMenuItem.vala

```diff
@@ -93,6 +93,9 @@ public class Network.VpnMenuItem : Gtk.ListBoxRow {
                     critical ("An VPN is being connected but not active.");
                 }
                 break;
+            case State.CONNECTED_VPN:
+                hide_item (spinner);
```

Comment:
`hide_item` is called at the start of the `update` method, so the spinner should already be hidden when control reaches this point.

Reply:
I see. I think that the update method is being called too early as a result of the async operation ```    nm_client.activate_connection_async.begin``` being called in ```Widgets/VpnInterface.vala``` I am quite new to this so I am not too certain of a fix for this yet.